### PR TITLE
add mongodb support

### PR DESCRIPTION
**Pull request by acgotaku**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -13,15 +14,27 @@ config.load();
 const container = new Container();
 
 let store: PersistentStorage<number|string>;
+let DBStore: any;
```
Comment (lordfriend) [inline src/main.ts]: You should bind PersistentStorage type here instead of bind it later.

File: src/main.ts [hunk 2]
```diff
@@ -13,15 +14,27 @@ config.load();
...
+        DBStore = MongodbStore;
+        break;
+    default:
+        throw new Error('DB_MODE is not support yet');
+}
 
 switch (config.mode) {
     case ConfigManager.DMHY:
-        container.bind<PersistentStorage<number>>(TYPES.PersistentStorage).to(PostgresStore).inSingletonScope();
+        container.bind<PersistentStorage<number>>(TYPES.PersistentStorage).to(DBStore).inSingletonScope();
```
Comment (lordfriend) [inline src/main.ts]: This could move to the switch(config.dbMode) code block

File: src/storage/mongodb-store.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    public getItem(id: T): Promise<Item<T> | null> {
+        return undefined;
+    }
+
+    public hasItem(id: T): Promise<boolean> {
+        return undefined;
+    }
+
+    public async filterItemNotStored(ids: T[]): Promise<T[]> {
+        const checkIds = ids.map(id => this._findById(id));
```
Comment (lordfriend) [inline src/storage/mongodb-store.ts]: Could this query do batch mode in order to improve performance?

File: src/storage/mongodb-store.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        return Promise.resolve(true);
+    }
+
+    public async onEnd(): Promise<void> { 
+        await mongoose.disconnect();
+    }
+
+    public async onStart(): Promise<void> {
+        const config = ConfigManager.getInstance();
+        const itemSchema = new mongoose.Schema({
```
Comment (lordfriend) [inline src/storage/mongodb-store.ts]: Can we use the class defined in `entity` directory as schema? Define schema separately may cause problem for maintain schema.

File: src/main.ts [hunk 5]
```diff
@@ -13,15 +14,27 @@ config.load();
 const container = new Container();
 
 let store: PersistentStorage<number|string>;
+let DBStore: any;
+
+switch(config.dbMode) {
+    case ConfigManager.PG:
+        DBStore = PostgresStore;
```
Comment (lordfriend) [inline src/main.ts]: use IoC container to bind PostgresStore to PersistentStorage instead of replace variable.

File: src/main.ts (hunk 2 above)
Comment (acgotaku) [reply src/main.ts]: In `DMHY` mode we need bind PersistentStorage with `number`, in `BANGUMI_MOE` mode
we need bind `PersistentStorage` wtth `string`. So, we need add if else to resolve it ?

Comment (lordfriend) [reply src/main.ts]: Oh, I got it. then your solution is ok

**Pull request closed without merging**